### ISSUE TITLE
qgis_query_version() errors on Fedora (workaround included)

### ISSUE THREAD

**Issue by Nowosad** [paleolimbot/qgisprocess#117]
See my reprex below. I suspect this issue is similar to https://github.com/paleolimbot/qgisprocess/issues/103.

``` r
# remotes::install_github("paleolimbot/qgisprocess")
library(qgisprocess)
#> Error in abort_query_version(lines = lines) : 
#>   Output did not contain expected version information and was:
#> 
#> QGIS 3.28.2-Firenze 'Firenze' (exported)
#> QGIS code branchRelease 3.28
#> Qt version 5.15.7
#> Python version 3.11.1
#> GDAL/OGR version 3.5.2
#> PROJ version 9.0.1
#> EPSG Registry database version v10.064 (2022-05-19)
#> GEOS version 3.11.0-CAPI-1.17.0
#> SQLite version 3.40.0
#> OS Fedora Linux 37 (Thirty Seven)
#> The 'qgis_process' command-line utility was not found.
#> Please run `qgis_configure()` to fix this and rebuild the cache.
#> See its documentation if you need to preset the path of qgis_process.
qgis_configure()
#> getOption('qgisprocess.path') was not found.
#> Sys.getenv('R_QGISPROCESS_PATH') was not found.
#> Trying 'qgis_process' on PATH...
#> Success!
#> Error in `abort_query_version()`:
#> ! Output did not contain expected version information and was:
#> 
#> QGIS 3.28.2-Firenze 'Firenze' (exported)
#> QGIS code branchRelease 3.28
#> Qt version 5.15.7
#> Python version 3.11.1
#> GDAL/OGR version 3.5.2
#> PROJ version 9.0.1
#> EPSG Registry database version v10.064 (2022-05-19)
#> GEOS version 3.11.0-CAPI-1.17.0
#> SQLite version 3.40.0
#> OS Fedora Linux 37 (Thirty Seven)

#> Backtrace:
#>     ▆
#>  1. └─qgisprocess::qgis_configure()
#>  2.   ├─base::tryCatch(...)
#>  3.   │ └─base (local) tryCatchList(expr, classes, parentenv, handlers)
#>  4.   │   └─base (local) tryCatchOne(expr, names, parentenv, handlers[[1L]])
#>  5.   │     └─base (local) doTryCatch(return(expr), name, parentenv, handler)
#>  6.   └─qgisprocess::qgis_version(query = TRUE, quiet = quiet)
#>  7.     └─qgisprocess::qgis_query_version(quiet = quiet)
#>  8.       └─qgisprocess:::abort_query_version(lines = lines)
#>  9.         └─rlang::abort(...)
```

**Comment by Nowosad** [comment]
Hi @florisvdh -- this is a fairly new issue (not seen a few weeks/months ago). It happens for both, QGIS available by default when using `dnf` (3.26, I believe)  and the one you mentioned (dani/qgis; 3.28).

**Comment by florisvdh** [comment]
@Nowosad you can try with `remotes::install_github("paleolimbot/qgisprocess@fedora")` for now. I still consider it a workaround, until more insight is available in the possible, or acceptable, variations of the QGIS version string.

**Comment by Nowosad** [comment]
Thanks:

``` r
# remotes::install_github("paleolimbot/qgisprocess@fedora")
library(qgisprocess)
#> Using 'qgis_process' in the system PATH.
#> QGIS version: 3.28.2-Firenze
#> Configuration loaded from '~/.cache/R-qgisprocess/cache-0.0.0.9000.rds'
#> Run `qgis_configure(use_cached_data = TRUE)` to reload cache and get more details.
#> >>> If you need another installed QGIS version, run `qgis_configure()`;
#>     see its documentation if you need to preset the path of qgis_process.
#> - Using JSON for input serialization.
#> - Using JSON for output serialization.
qgis_configure()
#> getOption('qgisprocess.path') was not found.
#> Sys.getenv('R_QGISPROCESS_PATH') was not found.
#> Trying 'qgis_process' on PATH...
#> Success!
#> QGIS version: 3.28.2-Firenze
#> Saving configuration to '~/.cache/R-qgisprocess/cache-0.0.0.9000.rds'
#> Metadata of 653 algorithms queried and stored in cache.
#> Run `qgis_algorithms()` to see them.
#> - Using JSON for input serialization.
#> - Using JSON for output serialization.
```

**Comment by florisvdh** [comment]
> this is a fairly new issue

Yes, it is a side effect of detecting and making distinction between different development versions of QGIS (#115), i.e. QGIS compiled locally (Linux) or distributed daily builds (for Windows & macOS). In these cases, the commit hash is included in the reported version string in **qgisprocess**.

> It happens for both, QGIS available by default when using `dnf` (3.26, I believe) and the one you mentioned (dani/qgis; 3.28).

Thanks! I plan to ask whether 'not including the commit hash' can be changed at Fedora and brought in accordance with the other OSes, which seems more ideal. The only reference to a commit hash is currently the branch 'Release 3.28', but that is not a stable reference (that branch still evolves at the QGIS repo); at least a git tag should be used if not a commit hash.

Regardless of this, we should probably postpone the requirement for a commit hash in `qgis_query_version()` until it is confirmed that a development version is being used – only then **qgisprocess** will actually incorporate that commit hash in the version string. It would protect against this sort of string variations, at least for the stable releases.

**Comment by florisvdh** [comment]
See https://github.com/daniviga/QGIS/issues/3 for the Fedora side.
